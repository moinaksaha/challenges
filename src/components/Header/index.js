// @flow
import React from 'react';
import PropTypes from 'prop-types';

import styled from 'styled-components';

const HeaderWrapper = styled.div`
  padding: 40px 0 10px 0;
  background: #ffffff;
  transition: 0.3s;
  display: flex;
  flex-direction: column;
  align-items: center;

  @media (max-width: 500px) {
    flex-direction: column !important;

    ${({ stuck }) => stuck && `
      padding: 10px;
    `}

  }

  ${({ stuck }) => stuck && `
    flex-direction: row;
    background: #648cf3;
    padding: 15px;
    justify-content: space-between;
  `}

  
`;

const PageHeading = styled.div`
  font-size: 30px;
  color: #5c606f;

  ${({ stuck }) => stuck && `
    font-size: 20px;
    color: #ffffff;
    font-weight: bold;
  `}
`;

const TotalDonation = styled.div`
  text-align: center;
  padding: 10px 0;
  font-size: 14px;
  color: #648cf3;

  @media (max-width: 500px) {
    padding: 5px;
    font-size: 12px;
  }

  ${({ stuck }) => stuck && `
    padding: 0;
    color: #ffffff;
  `}
`;

const Header = ({ totalDonation, isStuck }) => {
  if (totalDonation) {
    let totalDonationBreakup = '';

    Object.keys(totalDonation).forEach(function(key,index) {
      totalDonationBreakup = (index > 0) ?
        `${totalDonationBreakup}, ${totalDonation[key]} ${key}`:
        `${totalDonation[key]} ${key}`
    });

    return (
      <HeaderWrapper stuck={isStuck}>
        <PageHeading stuck={isStuck}>
            Omise Tamboon React
        </PageHeading>
        {totalDonation &&
          <TotalDonation stuck={isStuck}>
            {`Total Donation Collected: ${totalDonationBreakup}`}
          </TotalDonation>
        }
      </HeaderWrapper>
    );
  }
  return null;
};

Header.propTypes = {
  totalDonation: PropTypes.object,
};

Header.defaultProps = {
  totalDonation: {},
};

export default Header;


