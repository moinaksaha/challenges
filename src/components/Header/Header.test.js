import React from 'react'
import renderer from 'react-test-renderer'
import Header from './index.js'

it('Header: default', () => {
  const component = renderer.create(<Header />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})