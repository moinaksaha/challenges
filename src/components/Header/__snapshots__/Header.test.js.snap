// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <Header /> Snapshot: Enzyme 1`] = `
<styled.div>
  <styled.div>
    Omise Tamboon React
  </styled.div>
  <styled.div>
    Total Donation Collected: 
  </styled.div>
</styled.div>
`;
